iothub/device/src/DeviceClient.cs

```diff
@@ -608,7 +608,25 @@ public void SetRetryPolicy(IRetryPolicy retryPolicy)
         /// <summary>
         /// Releases the unmanaged resources used by the DeviceClient and optionally disposes of the managed resources.
         /// </summary>
-        public void Dispose() => InternalClient?.Dispose();
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Releases the unmanaged resources used by the DeviceClient and allows for any derived class to override and
+        /// provide custom implementation.
+        /// </summary>
+        /// <param name="disposing">Setting to true will release both managed and unmanaged resources.</param>
```

Comment:
do you think it would be worth calling out that setting to "false" would only release the unmanaged resources?

Reply:
it is kind-of touched in the method description, so I wonder if adding it to the param description is useful or not.

Reply:
No harm calling it out. But my PR has passed so let me do it in another PR.